### PR TITLE
#168781831 travel admin manage accommodations

### PR DESCRIPTION
**Pull request by alexismajya**
#### What does this PR do?

- create an accommodation facility
- edits an accommodation facility
- views accommodations by `id` and`availability` and `all`
   - `get` route: `api/v1/accommodations/search` where `isAvailable=true` or `false`
   -  `get` route: `api/v1/accommodations` for `all accommodations`
   -  `get` route: `api/v1/accommodations/accommodationId` for getting an `accommodation by Id'`

#### Description of Task to be completed?

- refactor  accommodation middleware
- refactor check roles middleware
- refactor accommodation route 
- refactor accommodation controller (only `getAccommodations`)
- refactor accommodation tests for `get` route
- add migration file `add-column-isAvailable`

#### How should this be manually tested?

- clone the repository
- run `npm test`
- using `postman` the routes

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[Pivotal story](https://www.pivotaltracker.com/story/show/168781831)

##### View accommodation

Screen Shot 2019-11-03 at 10 00 04

##### Edit an accommodation

Screen Shot 2019-11-03 at 10 00 56

##### create an  accommodation

Screen Shot 2019-11-01 at 13 05 02

#### Questions:

### CODE REVIEW

File: src/database/migrations/20191030205040-add-available-accommodation.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+`use strict`;
+
+module.exports = {
+  up: function(queryInterface, Sequelize) {
+    return queryInterface.addColumn(
+      'Accommodations',
+      'status',
```
Comment (elchroy) [inline src/database/migrations/20191030205040-add-available-accommodation.js]: - Change the datatype to `Sequelize.BOOLEAN`, and the column name to `isAvailable`
- The default value should be `true` unless when changed by the host

File: src/database/models/Accommodations.js [hunk 2]
```diff
@@ -73,6 +73,7 @@ module.exports = (sequelize, DataTypes) => {
     cost: DataTypes.DOUBLE,
     services: DataTypes.STRING,
     amenities: DataTypes.STRING,
+    status:DataTypes.STRING,
```
Comment (elchroy) [inline src/database/models/Accommodations.js]: Let the changes reflect here in the model.

Comment (alexismajya) [reply src/database/models/Accommodations.js]: thank you @elchroy, I applied the changes

File: src/database/migrations/20191030205040-add-available-accommodation.js (hunk 1 above)
Comment (alexismajya) [reply src/database/migrations/20191030205040-add-available-accommodation.js]: Thank you, I applied the changes

File: src/controllers/accommodationController.js [hunk 3]
```diff
@@ -48,20 +52,41 @@ const Accommodation = {
...
-      const accommodations = await db.Accommodations.findAll({
-        where: { userid: userId },
-        include: [{ model: db.AccommodationImages, attributes: { exclude: ['id', 'accommodationid', 'createdAt', 'updatedAt'] } }],
+  async getAccommodations(req, res) {
+    const { id, isAvailable, address, name, cost, costIsLess, costIsGreater, userid } = req.query;
+    let accommodations = '';
+    const image = [{ model: db.AccommodationImages, attributes: { exclude: ['id', 'accommodationid', 'createdAt', 'updatedAt'] } }];
+    if (id) {
+      accommodations = await db.Accommodations.findOne({
+        where: { id }, include: image, });
```
Comment (elchroy) [inline src/controllers/accommodationController.js]: this should go into a new route `/accommodations/:accommodationId`

File: src/controllers/accommodationController.js [hunk 4]
```diff
@@ -48,20 +52,41 @@ const Accommodation = {
...
+    } else if (userid) {
+      accommodations = await db.Accommodations.findAll({
+        where: { userid }, include: image, });
+    } else if (address) {
+      accommodations = await db.Accommodations.findAll({
+        where: { address }, include: image, });
+    } else if (cost) {
+      accommodations = await db.Accommodations.findAll({
+        where: { cost }, include: image, });
+    } else if (costIsLess) {
```
Comment (elchroy) [inline src/controllers/accommodationController.js]: - Add one route for `GET`ting `/accommodations`
- For searching, add a route - `/accommodations/search`. For now, allow only searching by availability (`isAvailable`)

File: src/database/migrations/20191030205040-add-available-accommodation.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+`use strict`;
```
Comment (elchroy) [inline src/database/migrations/20191030205040-add-available-accommodation.js]: Change filename to `add-isAvailable-to-Accommodations`

File: src/database/models/Accommodations.js [hunk 6]
```diff
@@ -73,6 +73,7 @@ module.exports = (sequelize, DataTypes) => {
     cost: DataTypes.DOUBLE,
     services: DataTypes.STRING,
     amenities: DataTypes.STRING,
+    isAvailable:DataTypes.BOOLEAN,
```
Comment (elchroy) [inline src/database/models/Accommodations.js]: Let the default value of isAvailable also be true!

**Review by swaibat — APPROVED** [review]

**Review by lemoissonM — APPROVED** [review]

**Review by lemoissonM — CHANGES_REQUESTED** [review]
fix first travis

**Review by swaibat — COMMENTED** [review]
great job @alexismajya  but try to fix your travis

**Review by lemoissonM — APPROVED** [review]

**Pull request merged**